docs/providers/laravel-jwt.md

```diff
@@ -0,0 +1,89 @@
+# Laravel JWT
+
+This provider is for the [Laravel JWT](https://github.com/tymondesigns/jwt-auth). Please make sure to follow our guide on the route configuration on this page to avoid issues on the authentication.
+
+The source code of the [Laravel JWT](https://github.com/tymondesigns/jwt-auth) provider can be found below [here](https://github.com/nuxt-community/auth-module/blob/master/lib/providers/laraveljwt.js).
+
+
+## Usage
+
+```js
+auth: {
+  strategies: {
+      'laravel.jwt': {
+        url: '...insert base url...'
+        endpoints: {
+            ...
+            },
+        token: {
+                property: 'access_token',
+                maxAge: 3600
+            },
+        refreshToken: {
+              
+                required: false
+            },
+      },
+  }
+}
+```
+
+Anywhere in your application logic:
+
+```js
+this.$auth.loginWith('laravel.jwt')
+```
+
+💁 This provider is based on [Refresh Scheme](../schemes/refresh.md).
+
+### Your base `url` explained & route configuration
+
+Following the official [Laravel JWT docs](https://jwt-auth.readthedocs.io/en/develop/quick-start/#add-some-basic-authentication-routes) you will generate the following routes:
+
+```php
+
+Route::group([
+
+    'middleware' => 'api',
+    'prefix' => 'auth'
+
+], function ($router) {
+
+    Route::post('login', 'AuthController@login');
+    Route::post('logout', 'AuthController@logout');
+    Route::post('refresh', 'AuthController@refresh');
+    Route::post('user', 'AuthController@user');
```

Comment:
I was looking the docs of Laravel JWT package, and I found some issues on their examples.
Maybe we can show here a fixed version of their examples, what do you think? 😄 

First thing is that the user route should be a `get` method.

Second is that in `AuthController` example, refresh route shouldn't use `auth:api` middleware.
```php
/**
  * Create a new AuthController instance.
  *
  * @return void
  */
  public function __construct() {
    $this->middleware('auth:api', ['except' => ['login', 'refresh']]);
  }
```

I'll point another issue in a comment below.

Reply:
maybe in our example we don't wrap it with any middleware. Leaving the user to fix and decide. But I'll send a PR to the author's doc to amend.

Reply:
Ok :)

Reply:
Here is my proposal 
```php

Route::group([

    'middleware' => 'api',
    'prefix' => 'auth'

], function ($router) {

    Route::post('login', 'AuthController@login');
    Route::post('logout', 'AuthController@logout');
    Route::post('refresh', 'AuthController@refresh');
    Route::get('user', 'AuthController@user')->withoutMiddleware(['api']);

});

```

Reply:
Nice! Just change to without `auth:api`

Reply:
Maybe add `auth:api` to middleware too.

```php

Route::group([

    'middleware' => ['api', 'auth:api'],
    'prefix' => 'auth'

], function ($router) {

    Route::post('login', 'AuthController@login');
    Route::post('logout', 'AuthController@logout');
    Route::post('refresh', 'AuthController@refresh');
    Route::get('user', 'AuthController@user')->withoutMiddleware(['auth:api']);

});

```

Reply:
ahh ok sorry got it

Reply:
should i put it on the exclude the auth:api on the login aswell? Just to make it explicit? I also made a mistake in my example so i fixed it.

Reply:
Yes I think you can put on login as well. But for both login and refresh is without `auth:api` only
`->withoutMiddleware(['auth:api'])`

Reply:
I've also PR laravel jwt to fix the issue. On another note i think the state should have scheme specific state store as 1 user can have many logins such as facebook/google/linkedin. Allowing the app developer to have the ability access fb/google/linkedin info using the `$auth.user['schemename']` would be better.

```js
  setUser (user) {
    this.$storage.setState('user', user)
    this.check()
  }
```

Reply:
Maybe open an issue to discuss about this, what do you think? :)